Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1

```diff
@@ -744,19 +744,19 @@ function Get-AzureSqlDatabaseServerResourceId {
         [Object] [Parameter(Mandatory = $true)] $endpoint,
         [Object] [Parameter(Mandatory = $true)] $accessToken)
 
-    $serverType = "Microsoft.Sql/servers"
+    $apiVersion = "2019-06-01-preview"
     $subscriptionId = $endpoint.Data.SubscriptionId.ToLower()
 
-    Write-Verbose "[Azure Rest Call] Get Resource Groups"
+    Write-Verbose "[Azure Rest Call] Get Azure Sql Servers"
     $method = "GET"
-    $uri = "$($endpoint.Url)/subscriptions/$subscriptionId/resources?api-version=$apiVersion"
+    $uri = "$($endpoint.Url)/subscriptions/$subscriptionId/providers/Microsoft.Sql/servers?api-version=$apiVersion"
     $headers = @{Authorization = ("{0} {1}" -f $accessToken.token_type, $accessToken.access_token) }
 
     do {
         Write-Verbose "Fetching Resources from $uri"
         $ResourceDetails = Invoke-RestMethod -Uri $uri -Method $method -Headers $headers -ContentType $script:jsonContentType
         foreach ($resourceDetail in $ResourceDetails.Value) {
-            if ($resourceDetail.name -eq $serverName -and $resourceDetail.type -eq $serverType) {
+            if ($resourceDetail.name -eq $serverName) {
                 return $resourceDetail.id
             }
         }
```

Comment:
Update `AZ_NoValidResourceIdFound` string which takes serverType input in this 
https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/Common/VstsAzureRestHelpers_/Strings/resources.resjson/en-US/resources.resjson 
and 
https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/Common/VstsAzureRestHelpers_/module.json
this file

Reply:
Re-added the `$serverType = "Microsoft.Sql/servers"` variable in `function Get-AzureSqlDatabaseServerResourceId` to align with the value thrown in case of an error and avoid downstream implications.